### PR TITLE
Documented the methods of VRDisplay and improved documentation of various attributes

### PR DESCRIPTION
**Pull request by toji**

### CODE REVIEW

**Comment by toji** [comment]
@cvan, do you have a moment to look this pull request over?

File: index.bs [hunk 1]
```diff
@@ -166,6 +166,37 @@ The {{isPresenting}} attribute MUST return the {{VRDisplay}}'s presentation stat
...
+
+<dfn method for="VRDisplay">resetPose()</dfn>
+Reset the pose for the display, treating its current position and orientation as the "origin/zero" values. Future poses returned from {{getPose()}} and {{getImmediatePose()}} will describe positions relative to the {{VRDisplay}}'s position when {{resetPose()}} was last called and will treat the display's yaw when {{resetPose()}} was last called as the forward orientation. The {{VRDisplay}}'s reported roll and pitch do not change when {{resetPose()}} is called as they are relative to gravity. Calling {{resetPose()}} may change the {{sittingToStandingTransform}} matrix of the {{VRStageParameters}}.
+
+<dfn method for="VRDisplay">requestAnimationFrame()</dfn>
+Functionally equivalent to <a href="https://www.w3.org/TR/html51/webappapis.html#animation-frames">window.requestAnimationFrame</a> when the {{VRDisplay}} is not presenting. When the {{VRDisplay}} is presenting the callback is called at the native refresh rate of the {{VRDisplay}}.
```
Comment (cvan) [inline index.bs:184]: could define the link [above in the `anchors` definitions](https://github.com/MozVR/webvr-spec/blob/display-methods/index.bs#L17)

File: index.bs [hunk 2]
```diff
@@ -166,6 +166,37 @@ The {{isPresenting}} attribute MUST return the {{VRDisplay}}'s presentation stat
 <dfn attribute for="VRDisplay">capabilities</dfn>
 The {{capabilities}} attribute MUST return the {{VRDisplay}}'s {{VRDisplayCapabilities}} object, a dictionary of capabilities describing the {{VRDisplay}}.
 
+<dfn method for="VRDisplay">getEyeParameters()</dfn>
+Returns the current {{VREyeParameters}} for the given eye. The eye parameters MAY change at any time due to external factors, such as the user changing the IPD with hardware controls. As a result, it is recommended that these values be queried for each frame rather than cached.
+
+<dfn method for="VRDisplay">getPose()</dfn>
+Returns a {{VRPose}} describing the position, orientation, and acceleration of the {{VRDisplay}} that should be used when rendering the next frame of a scene. The User Agent MAY optionally use predictive techniques to estimate what the pose will be at the time that the next frame will be displayed to the user. The pose returned MUST NOT change until JavaScript has returned control to the browser.
```
Comment (cvan) [inline index.bs:175]: end with a period

File: index.bs [hunk 3]
```diff
@@ -314,47 +345,50 @@ values will always be greater than or equal to older values.
...
 * Positive X is to the user's right.
 * Positive Y is up.
 * Positive Z is behind the user.
 
-All positions are given relative to the identity orientation, a sitting space.
+All positions are given relative to the identity orientation, in sitting space.
 Transforming this point with {{VRStageParameters}}.{{sittingToStandingTransform}}</code>
-converts this to standing space. May be null if the sensor is incapable of
-providing positional data. When not null MUST be a three-element array.
+converts this to standing space. MAY be null if the sensor is incapable of
```
Comment (cvan) [inline index.bs]: `null` is `NULL` elsewhere

File: index.bs [hunk 4]
```diff
@@ -314,47 +345,50 @@ values will always be greater than or equal to older values.
...
+providing positional data. User agents MAY provide emulated position values
+through techniques like neck modeling, but when doing so SHOULD report
+{{VRDisplayCapabilities}}.{{hasPosition}} as false. When not null MUST be a
+three-element array.
 
 <dfn attribute for="VRPose">linearVelocity</dfn>
-Linear velocity of the sensor at {{timestamp}}. May be null if the sensor is
-incapable of providing linear velocity. When not null MUST be a three-element
-array.
+Linear velocity of the sensor at {{timestamp}} meters per second. MAY be null if
```
Comment (cvan) [inline index.bs]: `null` again

File: index.bs [hunk 5]
```diff
@@ -314,47 +345,50 @@ values will always be greater than or equal to older values.
...
+the sensor is incapable of providing linear velocity. When not null MUST be a
```
Comment (cvan) [inline index.bs]: `null` again

File: index.bs [hunk 6]
```diff
@@ -424,7 +458,7 @@ interface VREyeParameters {
 ### Attributes ### {#vreyeparameters-attributes}
 
 <dfn attribute for="VREyeParameters">offset</dfn>
-Offset from the center of the user's head to the eye in meters. This value SHOULD represent the user's interpupillary distance (IPD), but may also represent the distance from the center point of the headset to the center point of the lens for the given eye. Values for the left eye MUST be negative; values for the right eye MUST be positive.
+Offset from the center point between the users eyes to the center of the eye in meters. This value SHOULD represent half of the user's interpupillary distance (IPD), but MAY also represent the distance from the center point of the headset to the center point of the lens for the given eye. Values for the left eye MUST be negative; values for the right eye MUST be positive.
```
Comment (cvan) [inline index.bs:463]: `users eyes` -> `user's eyes`

File: index.bs [hunk 7]
```diff
@@ -166,6 +166,37 @@ The {{isPresenting}} attribute MUST return the {{VRDisplay}}'s presentation stat
...
+
+<dfn method for="VRDisplay">getImmediatePose()</dfn>
+Returns a {{VRPose}} describing the position, orientation, and acceleration of the {{VRDisplay}} at the time when the call was made. The User Agent SHOULD NOT use predictive techniques when constructing the pose.
+
+<dfn method for="VRDisplay">resetPose()</dfn>
+Reset the pose for the display, treating its current position and orientation as the "origin/zero" values. Future poses returned from {{getPose()}} and {{getImmediatePose()}} will describe positions relative to the {{VRDisplay}}'s position when {{resetPose()}} was last called and will treat the display's yaw when {{resetPose()}} was last called as the forward orientation. The {{VRDisplay}}'s reported roll and pitch do not change when {{resetPose()}} is called as they are relative to gravity. Calling {{resetPose()}} may change the {{sittingToStandingTransform}} matrix of the {{VRStageParameters}}.
```
Comment (cvan) [inline index.bs]: can you change `display` to `{{VRDisplay}}`?

File: index.bs [hunk 8]
```diff
@@ -314,47 +345,50 @@ values will always be greater than or equal to older values.
...
+in meters from an origin point, which is either the position the sensor was
+first read at or the position of the sensor at the point that {{resetPose()}}
 was last called. The coordinate system uses these axis definitions:
 
 * Positive X is to the user's right.
 * Positive Y is up.
 * Positive Z is behind the user.
 
-All positions are given relative to the identity orientation, a sitting space.
+All positions are given relative to the identity orientation, in sitting space.
```
Comment (cvan) [inline index.bs]: you can remove the comma

File: index.bs [hunk 9]
```diff
@@ -314,47 +345,50 @@ values will always be greater than or equal to older values.
...
 Transforming this point with {{VRStageParameters}}.{{sittingToStandingTransform}}</code>
-converts this to standing space. May be null if the sensor is incapable of
-providing positional data. When not null MUST be a three-element array.
+converts this to standing space. MAY be null if the sensor is incapable of
+providing positional data. User agents MAY provide emulated position values
+through techniques like neck modeling, but when doing so SHOULD report
```
Comment (cvan) [inline index.bs]: `like` should be `such as`

**Comment by cvan** [comment]
r+wc everything else looks good. if you could just make those changes and rebuild, I think we're good here. thanks!

**Comment by toji** [comment]
I'll see if I can address you comments this weekend. Thanks!

**Pull request merged**